openassessment/assessment/api/teams.py

```diff
@@ -0,0 +1,363 @@
+    team submission.
+
+    Note that the staff workflow begins things in on_init() instead of
+    on_start(), because staff shoud be able to access the submission
+    regardless of which state the workflow is currently in.
+
+    Args:
+        team_submission_uuid (str): The team submission associated with this workflow.
+
+    Returns:
+        None
+
+    Raises:
+        StaffAssessmentInternalError: Raised when there is an internal error
+            creating the Workflow.
+    """
+    try:
+        team_submission = team_submissions_api.get_team_submission(team_submission_uuid)
+        TeamStaffWorkflow.objects.get_or_create(
+            course_id=team_submission['course_id'],
+            item_id=team_submission['item_id'],
+            team_submission_uuid=team_submission_uuid
+        )
+    except DatabaseError:
+        error_message = (
+            "An internal error occurred while creating a new team staff workflow for team submission {}"
+        ).format(team_submission_uuid)
+        logger.exception(error_message)
+        raise StaffAssessmentInternalError(error_message)
+
+
+def on_cancel(team_submission_uuid):
+    """
+    Cancel the team staff workflow for submission.
+
+    Sets the cancelled_at field in team staff workflow.
+
+    Args:
+        team_submission_uuid (str): The team_submission UUID associated with this workflow.
+
+    Returns:
+        None
+
+    """
+    try:
+        workflow = TeamStaffWorkflow.objects.get(team_submission_uuid=team_submission_uuid)
+        workflow.cancelled_at = now()
+        workflow.save(update_fields=['cancelled_at'])
+    except TeamStaffWorkflow.DoesNotExist:
+        # If we can't find a workflow, then we don't have to do anything to
+        # cancel it.
+        pass
+    except DatabaseError:
+        error_message = (
+            "An internal error occurred while cancelling the team staff workflow for submission {}"
+        ).format(team_submission_uuid)
+        logger.exception(error_message)
+        raise StaffAssessmentInternalError(error_message)
+
+
+def get_score(team_submission_uuid, staff_requirements):  # pylint: disable=unused-argument
+    """
+    Generate a score based on a completed assessment for the given team submission.
+    If no assessment has been completed for this submission, this will return
+    None.
+
+    Args:
+        team_submission_uuid (str): The UUID for the submission to get a score for.
+        staff_requirements (dict): Not used.
+
+    Returns:
+        A dictionary with the points earned, points possible,
+        contributing_assessments, and staff_id information.
+
+    """
+    assessment = get_latest_staff_assessment(team_submission_uuid)
+    if not assessment:
+        return None
+
+    return {
+        "points_earned": assessment["points_earned"],
+        "points_possible": assessment["points_possible"],
+        "contributing_assessments": [assessment["id"]],
```

Comment:
for contributing_assessments - we appear to always return a single item. Is their a plan to return more?

Reply:
Good question! The data structure that the staff API produces has a `contributing_assessments` array with only one assessment, so I kept the structure the same here.

Why is it an array? Possibly to allow showing of assessments and override history in the future, but this behavior hasn't been implemented in the staff API.

Reply:
likelihood is for model consistency with peer assessments where multiple assessments are expected.